index.js

```diff
@@ -1,25 +1,32 @@
 var os = require('os');
 
-var ipv4 = function() {
-	var interfaces = os.networkInterfaces()
-	for (var i in interfaces) {
-		for (var j = interfaces[i].length-1; j >= 0; j--) {
-			var face = interfaces[i][j]
-			if (!face.internal && face.family === 'IPv4') return face.address
-		}
-	}
-	return '127.0.0.1'
+function pickInterface(interfaces, family) {
+    for (var i in interfaces) {
+        for (var j = interfaces[i].length-1; j >= 0; j--) {
+            var face = interfaces[i][j]
+            if (!face.internal && face.family === family) return face.address
+        }
+    }
+    return family == 'IPv4' ? '127.0.0.1' : '::1'
 }
 
-var ipv6 = function() {
-  var interfaces = os.networkInterfaces()
-  for (var i in interfaces) {
-    for (var j = interfaces[i].length-1; j >= 0; j--) {
-      var face = interfaces[i][j]
-      if (!face.internal && face.family === 'IPv6') return face.address
-    }
-  }
-  return '::1'
+function reduceInterfaces(interfaces, iface) {
+    return Object.keys(interfaces).reduce((ifaces, _iface) => {
```

Comment:
could you change this to a classic function? i still wanna support 0.10. the identation seems a bit off as well